app/src/main/java/fr/neamar/kiss/dataprovider/AppProvider.java

```diff
@@ -19,30 +20,68 @@ public void reload() {
         ArrayList<Pojo> records = new ArrayList<>();
 
         int relevance;
-        int matchPositionStart;
-        int matchPositionEnd;
         String appNameNormalized;
 
-        final String queryWithSpace = " " + query;
         for (Pojo pojo : pojos) {
             relevance = 0;
             appNameNormalized = pojo.nameNormalized;
 
-            matchPositionEnd = 0;
-            if (appNameNormalized.startsWith(query)) {
-                relevance = 100;
-                matchPositionStart = 0;
-                matchPositionEnd = query.length();
-            } else if ((matchPositionStart = appNameNormalized.indexOf(queryWithSpace)) > -1) {
-                relevance = 50;
-                matchPositionEnd = matchPositionStart + queryWithSpace.length();
-            } else if ((matchPositionStart = appNameNormalized.indexOf(query)) > -1) {
-                relevance = 1;
-                matchPositionEnd = matchPositionStart + query.length();
+            ArrayList<Pair<Integer, Integer>> matchPositions = new ArrayList<>();
```

Comment:
my Java is a bit rusty, but I think we're instantiating an ArrayList for each Pojo for every character, which is probably more than we really need. Maybe only init the ArrayList on first match?

(according to http://programmers.stackexchange.com/questions/149563/should-we-avoid-object-creation-in-java, this is probably no big deal, but still feels a bit overkill to me?)